docs/upgrading-to-v0.10.md

```diff
@@ -0,0 +1,75 @@
+# Upgrading to Ark v0.10
+
+## Overview
+
+Ark v0.10 includes a number of breaking changes. Below, we outline what those changes are, and what steps you should take to ensure
+a successful upgrade from prior versions of Ark.
+
+## Breaking Changes
+
+### Switch from Config CRD to BackupStorageLocation, VolumeSnapshotLocation, and server flags
+
+Prior to v0.10, Ark used a `Config` CRD to capture information about your backup storage and persistent volume providers, as well
+some miscellaneous Ark settings. In v0.10, we've eliminated this CRD and replaced it with:
+
+- A [BackupStorageLocation][1] CRD to capture information about where to store your backups
+- A [VolumeSnapshotLocation][2] CRD to capture information about where to store your persistent volume snapshots
+- Command-line flags for the `ark server` command (run by your Ark deployment) to capture miscellaneous Ark settings
+
+When upgrading to v0.10, you'll need to transfer the configuration information that you currently have in the `Config` CRD
+into the above. We'll cover exactly how to do this below.
+
+### Reorganization of data in object storage
+
+We've made [changes to the layout of data stored in object storage][3] for simplicity and extensibility. You'll need to
+rearrange any pre-v0.10 data as part of the upgrade. We've provided a script to help with this.
+
+## Step-by-Step Upgrade Instructions
+
+1. Scale down your existing Ark deployment:
+```bash
+kubectl scale -n heptio-ark deploy/ark --replicas 0
+```
+
+2. Re-apply the `00-prereqs.yaml` file to create new CRDs:
+```bash
+kubectl apply -f examples/common/00-prereqs.yaml
+```
+
+3. Create a [BackupStorageLocation][1] resource based on the examples provided in the `examples/` directory for your platform, using information from
+the existing `Config` resource as necessary.
+
+4. Create a [VolumeSnapshotLocation][2] resource based on the examples provided in the `examples/` directory for your platform, using information from
+the existing `Config` resource as necessary.
+
+5. Perform the one-time object storage migration detailed [here][3].
+
+6. In your Ark deployment YAML (see the `examples/` directory for samples), specify any non-default settings as args to the `ark server` command:
+
+| Flag | Default Value |
+| ---- | ------------- |
+| `--backup-sync-period` | 1m |
+| `--restic-timeout` | 1h |
+| `--restore-only` | false |
+
+7. In your Ark deployment YAML, define your default `BackupStorageLocation` and `VolumeSnapshotLocations`  as args to the `ark server` command:
+
+| Flag | Default Value | Description |
+| ---- | ------------- | ----------- |
+| `--default-backup-storage-location` | "default" | name of the backup storage location that should be used by default for backups |
```

Comment:
Technically this is redundant except if they change the name to something other than `default`. If you think the suggestion I made above makes sense, maybe change the name here to something else, idk, maybe `new-location`.

Reply:
I was referring to the name `default`, since if you don't specify it it is the name ark server gives the default bsl. So naming it `default` is redundant.

Reply:
The server doesn't create or name a BSL. That's name of the BSL it looks for - that CR still has to be created.

Reply:
This was very confusing to me. I'm just saying, *now* I grasp this, but it took a lot of effort in trying to wrap my head around it. I suspect the language we are using could lead to understanding that we are creating a bsl, even tho we know this is not so.

Reply:
`That's name of the BSL it looks for` -> I really like this language.